manifests/windows/agent6.pp

```diff
@@ -31,15 +36,21 @@
     }
 
     exec { 'validate':
-      command   => "\$blacklist = '928b00d2f952219732cda9ae0515351b15f9b9c1ea1d546738f9dc0fda70c336','78b2bb2b231bcc185eb73dd367bfb6cb8a5d45ba93a46a7890fd607dc9188194';\$fileStream = [system.io.file]::openread('${msi_full_path}'); \$hasher = [System.Security.Cryptography.HashAlgorithm]::create('sha256'); \$hash = \$hasher.ComputeHash(\$fileStream); \$fileStream.close(); \$fileStream.dispose();\$hexhash = [system.bitconverter]::tostring(\$hash).ToLower().replace('-','');if (\$hexhash -match \$blacklist) { Exit 1 }",
+      command   => "\$blacklist = '928b00d2f952219732cda9ae0515351b15f9b9c1ea1d546738f9dc0fda70c336','78b2bb2b231bcc185eb73dd367bfb6cb8a5d45ba93a46a7890fd607dc9188194';\$fileStream = [system.io.file]::openread('${msi_full_path}'); \$hasher = [System.Security.Cryptography.HashAlgorithm]::create('sha256'); \$hash = \$hasher.ComputeHash(\$fileStream); \$fileStream.close(); \$fileStream.dispose();\$hexhash = [system.bitconverter]::tostring(\$hash).ToLower().replace('-','');if (\$blacklist.Contains(\$hexhash)) { Exit 1 }",
       provider  => 'powershell',
       logoutput => 'on_failure',
       require   => File['installer'],
       notify    => Package[$datadog_agent::params::package_name]
     }
 
+    if $agent_version == 'latest' {
+      $ensure_version = 'installed'
+    } else {
+      $ensure_version = $agent_version
+    }
+
     package { $datadog_agent::params::package_name:
-      ensure          => installed,
+      ensure          => $ensure_version,
```

Comment:
Does this actually do something on windows?

Reply:
If you set it to "installed", it won't upgrade it as long as it's installed. When you pass a version, it checks the installed version vs version the MSI and installs if they are different.